lib/nav.py

```diff
@@ -36,3 +56,116 @@ def force_dir(dir, n=1):
 			if ACTIONS[OPS.MOVE][dir]():
 				break
 			ACTIONS[OPS.DIG][dir]()
+
+# move n blocks
+def dir(dir, n=1):
+	for i in range(n):
+		fuel.fuel()
+		if not ACTIONS[OPS.MOVE][dir]():
+			return False
+	return True
+
+# turn a by a given direction
+def turn(turn_dir):
+	ACTIONS[OPS.TURN][turn_dir]()
+
+class Navigator:
+	def __init__(self, location=[0,0,0], direction=CARDINALS.NORTH):
+		self.location=location
+		self.direction=direction
+	### GETTERS
+	def get_location(self):
+		return self.location.copy()
+	def get_direction(self):
+		return CARDINALS(self.direction)
```

Comment:
are these getters really needed?

In python it's more standard to just directly access the fields themselves

Reply:
Done